drivers/shared/src/vic_time.c

```diff
@@ -0,0 +1,794 @@
+/******************************************************************************
+ * @section DESCRIPTION
+ *
+ * VIC time and calendar module
+ *
+ * Ported from https://github.com/Unidata/netcdf4-python/blob/master/netcdftime/netcdftime.py
+ *
+ * @section LICENSE
+ *
+ * The Variable Infiltration Capacity (VIC) macroscale hydrological model
+ * Copyright (C) 2014 The Land Surface Hydrology Group, Department of Civil
+ * and Environmental Engineering, University of Washington.
+ *
+ * The VIC model is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *****************************************************************************/
+
+#include <math.h>
+
+#include <vic_def.h>
+#include <vic_run.h>
+#include <vic_driver_shared.h>
+
+// FUNCTION PROTOTYPES
+double _FractionalDayFromDate(dmy_struct *date);
+double _JulianDayFromDate(dmy_struct *date, unsigned short calendar);
+double _NoLeapDayFromDate(dmy_struct *date);
+double _AllLeapFromDate(dmy_struct *date);
+double _360DayFromDate(dmy_struct *date);
+void _DMYJulianDay(double julian, unsigned short calendar, dmy_struct *dmy);
+void _DMYNoLeapDay(double julian, dmy_struct *dmy);
+void _DMYAllLeap(double julian, dmy_struct *dmy);
+void _DMY360Day(double julian, dmy_struct *dmy);
+bool _leap_year(unsigned short year, unsigned short calendar);
+
+/******************************************************************************
+ * @brief   Get fractional day of month from dmy structure.
+ * @return  Fractional Day.
+ *****************************************************************************/
+double
+_FractionalDayFromDate(dmy_struct *date)
+{
+    double day;
+
+    day = (double)date->day + (double)date->dayseconds / (double)SEC_PER_DAY;
```

Comment:
space between typecast and variable. See earlier discussion on using `#define`d variables.